pkg/client/xcp.go

```diff
@@ -10,11 +10,39 @@ import (
 )
 
 const (
-	metaGitRemote string = "SOLUBLE_METADATA_GIT_REMOTE"
-	metaGitBranch string = "SOLUBLE_METADATA_GIT_BRANCH"
-	metaHostname  string = "SOLUBLE_METADATA_HOSTNAME"
+	// MetaGitBranch is the current git branch
+	MetaGitBranch    string = "SOLUBLE_METADATA_GIT_BRANCH"
+	metaGitBranchCmd string = "git rev-parse --abbrev-ref HEAD"
+
+	// MetaGitCommit is the current git commit, `git rev-parse HEAD`
+	MetaGitCommit    string = "SOLUBLE_METADATA_GIT_COMMIT"
+	metaGitCommitCmd string = "git rev-parse HEAD"
+
+	// MetaGitCommitShort is the short version of the git commit.
+	MetaGitCommitShort    string = "SOLUBLE_METADATA_GIT_COMMIT_SHORT"
+	metaGitCommitShortCmd string = "git rev-parse --short HEAD"
+
+	// MetaGitTag is the current git tag for HEAD
+	MetaGitTag    string = "SOLUBLE_METADATA_GIT_TAG"
+	metaGitTagCmd string = "git describe --tags"
```

Comment:
Introducing constants for the commands makes this harder to read

Reply:
really? I found it much easier to understand what the commands do by seeing the underlying commands.